src/components/cl/ucc_cl.h

```diff
@@ -80,8 +80,11 @@ typedef struct ucc_cl_context {
 
 UCC_CLASS_DECLARE(ucc_cl_context_t, ucc_cl_lib_t *);
 
-#define UCC_CL_IFACE_EXT(_NAME) .super.type = UCC_CL_ ## _NAME,
+typedef struct ucc_cl_lib_attr {
+    ucc_base_attr_t super;
+    int tls;
```

Comment:
How do you set the required TLs? Does the CL and TLs have some kind of identifiers?

Reply:
yes CLs and TLs have both have enums. Some identifier is needed anyways. Originally we thought that for TLs we can use "char *". But since any TL (even vendor specific black box) requires integration into CL then there is no problem adding new enum id for any TL. And this way it is much more convenient.